### PR TITLE
bonfire: init at 0.0.7

### PR DESCRIPTION
**Pull request by womfoo**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/bonfire/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, python3Packages, fetchFromGitHub }:
+
+with python3Packages;
+
+buildPythonApplication rec {
+  version = "0.0.7-20170119";
```
Comment (vyp) [inline pkgs/tools/misc/bonfire/default.nix]: In this case version should just be `20170119`, unless this versioning scheme is recommended somewhere?

Comment (womfoo) [reply pkgs/tools/misc/bonfire/default.nix]: Thanks for reviewing @vyp . Rebased changes with just 20170119.

Comment (orivej) [reply pkgs/tools/misc/bonfire/default.nix]: Actually it should be `2017-01-19`: https://nixos.org/nixpkgs/manual/#sec-package-naming

**Comment by orivej** [comment]
@womfoo The package install some extraneous files:
```
$ tree result/
├── AUTHORS.rst
├── bin
│   └── bonfire
…
├── LICENSE.txt
├── nix-support
│   └── propagated-native-build-inputs
├── README.rst
└── requirements.txt
```
Can you fix that?

**Comment by womfoo** [comment]
@orivej just rebased the changes to remove extraneous files.

I also tried updating the version to `2017-01-19` and `unstable-2017-01-19` but pip fails with these.

```
/nix/store/9b4d3vb92z7b4ydxdk1zg9slvcfdgaxm-python3.6-bootstrapped-pip-9.0.1/lib/python3.6/site-packages/setuptools/dist.py:360: UserWarning: The version specified ('unstable-2017-01-19') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
  "details." % self.metadata.version
```

**Comment by vyp** [comment]
@womfoo try putting "unstable" in `pname` so that pip gets the date as the version:

```
version = "2017-01-19";
pname = "bonfire-unstable";
```

**Comment by vyp** [comment]
But that would mean `repo` for `src` wouldn't work. Just put "unstable" in `name` then: `name = "${pname}-unstable-${version}";`.

**Comment by womfoo** [comment]
Moving "unstable" in the `version` did fix it. Thanks @vyp

**Pull request merged**